client/overview/task-list/dispute-tasks.ts

```diff
@@ -0,0 +1,227 @@
+/**
+ * External dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import moment from 'moment';
+import { getHistory } from '@woocommerce/navigation';
+
+/**
+ * Internal dependencies
+ */
+import type { TaskItemProps } from './types';
+import type { CachedDispute } from 'wcpay/types/disputes';
+import { formatCurrency } from 'wcpay/utils/currency';
+import { getAdminUrl } from 'wcpay/utils';
+import wcpayTracks from 'wcpay/tracks';
+
+/**
+ * Returns true if the dispute is due within the specified number of days.
+ * Returns false if the dispute is not due within the specified number of days
+ * or if the due_by value is an empty string.
+ *
+ * @param {CachedDispute} dispute - The dispute to check.
+ * @param {number} days - The number of days to check. Defaults to 1.
+ *
+ * @return {boolean} True if the dispute is due within the specified number of days.
+ */
+const isDueWithin = ( dispute: CachedDispute, days: number ) => {
+	if ( dispute.due_by === '' ) {
+		return false;
+	}
+	const now = moment();
+	const dueBy = moment( dispute.due_by );
+	return (
+		dueBy.diff( now, 'hours' ) > 0 &&
+		dueBy.diff( now, 'hours' ) <= 24 * days
+	);
+};
+
+/**
+ * Returns an array of disputes that are due within the specified number of days.
+ *
+ * @param {CachedDispute[]} activeDisputes - The active disputes to filter.
+ * @param {number} days - The number of days to check.
+ *
+ * @return {CachedDispute[]} The disputes that are due within the specified number of days.
+ */
+export const getDisputesDueWithinDays = (
+	activeDisputes: CachedDispute[],
+	days: number
+): CachedDispute[] =>
+	activeDisputes.filter( ( dispute ) => isDueWithin( dispute, days ) );
+
+export const getDisputeResolutionTask = (
+	/**
+	 * Active disputes (awaiting a response) to generate the notice string for.
+	 */
+	activeDisputes: CachedDispute[]
+): TaskItemProps | null => {
+	// Create a new array and sort by `due_by` ascending.
+	activeDisputes = [ ...activeDisputes ]
+		.filter( ( dispute ) => dispute.due_by !== '' )
+		.sort( ( a, b ) => moment( a.due_by ).diff( moment( b.due_by ) ) );
+
+	const activeDisputeCount = activeDisputes.length;
+
+	if ( activeDisputeCount === 0 ) {
+		return null;
+	}
+
+	const handleClick = () => {
+		wcpayTracks.recordEvent( wcpayTracks.events.OVERVIEW_TASK_CLICK, {
+			task: 'dispute-resolution-task',
+			activeDisputeCount,
```

Comment:
When I grep `eventprop_name` in `tracks-events-registration` repo, looks like the convention for prop name is all lowercase with underscore. I don't think it's a big deal but for consistency sake, maybe we should use `active_dispute_count` here.

```suggestion
			active_dispute_count: activeDisputeCount,
```

Reply:
Yes! Thanks, the FG mentions lowercase + underscores are required for event props.

Reply:
I've committed your suggestion in a41fb14